flang/runtime/file.h

```diff
@@ -38,7 +44,7 @@ class OpenFile {
   bool mayPosition() const { return mayPosition_; }
   void set_mayPosition(bool yes) { mayPosition_ = yes; }
   FileOffset position() const { return position_; }
-  bool isTerminal() const { return isTerminal_; }
+  bool isTerminal() const { return ::isatty(fd_) == 1; }
```

Comment:
This will cause `isatty()` to be called (to return the same value) every time it's needed.  It's better to capture it once in the data member `isTerminal_`, I think.

Reply:
> This will cause `isatty()` to be called (to return the same value) every time it's needed. It's better to capture it once in the data member `isTerminal_`, I think.

I agree, though I am not sure where exactly I can put it such that it is called once and the correct file descriptor is checked.